### PR TITLE
bug fix issue 1204: On ts.del update lastTimestamp and lastValue cahc…

### PR DESCRIPTION
**Pull request by OfirMos**
…hed in the series if it was changed

fixing this issue: https://github.com/RedisTimeSeries/RedisTimeSeries/issues/1204

### CODE REVIEW

File: src/tsdb.c [hunk 1]
```diff
@@ -872,6 +872,20 @@ size_t SeriesDelRange(Series *series, timestamp_t start_ts, timestamp_t end_ts)
...
     CompactionDelRange(series, start_ts, end_ts);
+
+    // Check if last timestamp deleted
+    if (end_ts >= series->lastTimestamp) {
+        iter = RedisModule_DictIteratorStartC(series->chunks, "$", NULL, 0);
+        currentKey = RedisModule_DictNextC(iter, &keyLen, (void *)&currentChunk);
+        if (!currentKey) {
+            // No samples in the series
+            series->lastTimestamp = 0;
+            series->lastValue = 0;
```
Comment (filipecosta90) [inline src/tsdb.c:883]: @OfirMos can we extend the test on ts_delete to include also the case in which the series is now empty after the delete?

File: src/chunk.c [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
 #include "rmutil/alloc.h"
 
+#include <assert.h>
```
Comment (MeirShpilraien) [inline src/chunk.c:12]: Please use RedisModule_Assert

File: src/series_iterator.c [hunk 3]
```diff
@@ -65,7 +67,9 @@ void SeriesIteratorClose(AbstractIterator *iterator) {
 EnrichedChunk *SeriesIteratorGetNextChunk(AbstractIterator *abstractIterator) {
     SeriesIterator *iter = (SeriesIterator *)abstractIterator;
     Chunk_t *curChunk = iter->currentChunk;
-    if (!curChunk) {
+    if (!curChunk || iter->series->funcs->GetNumOfSample(curChunk) == 0) {
+        assert(iter->series->totalSamples == 0);
+        // If there is an empty chunk it means the whole series is empty
```
Comment (danni-m) [inline src/series_iterator.c:72]: what if the last chunk is empty?

File: src/tsdb.c [hunk 4]
```diff
@@ -859,10 +862,23 @@ size_t SeriesDelRange(Series *series, timestamp_t start_ts, timestamp_t end_ts)
...
         deletedSamples += funcs->GetNumOfSample(currentChunk);
         funcs->FreeChunk(currentChunk);
 
+        if (isLastChunkDeleted) {
+            Chunk_t *lastChunk;
+            RedisModuleDictIter *lastChunkIter =
+                RedisModule_DictIteratorStartC(series->chunks, "$", NULL, 0);
+            RedisModule_DictNextC(lastChunkIter, NULL, (void *)&lastChunk);
+            series->lastChunk = lastChunk;
+            RedisModule_DictIteratorStop(lastChunkIter);
```
Comment (danni-m) [inline src/tsdb.c:879]: Can we add a test to cover this?

File: src/tsdb.c [hunk 5]
```diff
@@ -859,10 +862,23 @@ size_t SeriesDelRange(Series *series, timestamp_t start_ts, timestamp_t end_ts)
             continue;
         }
 
+        bool isLastChunkDeleted = false;
+        if (currentChunk == series->lastChunk) {
+            isLastChunkDeleted = true;
```
Comment (danni-m) [inline src/tsdb.c:867]: Missing test?

**Review by danni-m — DISMISSED** [review]

**Review by danni-m — APPROVED** [review]

**Pull request merged**